apis/core.oam.dev/common/types.go

```diff
@@ -476,6 +481,48 @@ const (
 	WorkflowResourceCreator ResourceCreatorRole = "workflow"
 )
 
+// OAMObjectReference defines the object reference for an oam resource
+type OAMObjectReference struct {
+	Component string `json:"component,omitempty"`
+	Trait     string `json:"trait,omitempty"`
+	Env       string `json:"env,omitempty"`
+}
+
+// Equal check if two references are equal
+func (in OAMObjectReference) Equal(r OAMObjectReference) bool {
+	return in.Component == r.Component && in.Trait == r.Trait && in.Env == r.Env
```

Comment:
Can `reflect.DeepEqual` do the comparasion?

Reply:
Yes, there are several unit-tests can ensure that.

Reply:
Yes. It can. It is also possible to do the equality check with `reflect.DeepEqual`.